components/script/dom/canvasrenderingcontext2d.rs

```diff
@@ -543,18 +542,11 @@ impl CanvasRenderingContext2D {
                         Some(ref canvas) => &**canvas,
                     };
 
-                    let window = window_from_node(canvas);
+                    let canvas_element = canvas.upcast::<Element>();
 
-                    let style = window.GetComputedStyle(canvas.upcast(), None);
-
-                    let element_not_rendered =
-                        !canvas.upcast::<Node>().is_in_doc() ||
-                        style.GetPropertyValue(DOMString::from("display")) == "none";
-
-                    if element_not_rendered {
-                        Ok(RGBA::new(0, 0, 0, 255))
-                    } else {
-                        self.parse_color(&style.GetPropertyValue(DOMString::from("color")))
+                    match canvas_element.style() {
+                        Some(ref s) if canvas_element.has_css_layout_box() => Ok(s.get_color().color),
```

Comment:
The `has_css_layout_box` check whole new style query, which is somewhat unfortunate, but can be fixed in a followup I guess.

Thanks a lot!

Reply:
Man, didn't realize that style queries were expensive. Thanks for the help as well! I've learnt quite a bit from this PR.